Modules/Test/classes/class.ilTestExportAbstract.php

```diff
@@ -0,0 +1,247 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+/**
+ * @author Fabian Helfer <fhelfer@databay.de>
+ */
+abstract class ilTestExportAbstract
+{
+    protected ilObjTest $test_obj;
+    protected ilLanguage $lng;
+    protected string $filterby;
+    protected string $filtertext;
```

Comment:
IMO these two parameters are semantically weak. What is a `filterby`, what is a `filtertext`? What are valid criteria to filter the export by? I suggest do create explicit classes and replace this primitive string parameters.